app/src/main/java/ingsw/group1/findmyphone/cryptography/SMSCipher.java

```diff
@@ -0,0 +1,145 @@
+        return new SMSMessage(peerOfMessage, decryptedMessage);
+    }
+
+    /**
+     * Encrypts the string using a xor encryption with the key.
+     *
+     * @param stringToCipher The string to be encrypted.
+     * @param key            The Key used for encryption.
+     * @return The encrypted String.
+     */
+    public static String encrypt(String stringToCipher, String key) {
+        int[] strInt = SMSCipher.fromStringToAscii(stringToCipher);
+        int[] keyInt = SMSCipher.fromStringToAscii(key);
+        StringBuilder encryptedStr = new StringBuilder();
+        for (int i = 0; i < strInt.length; i++) {
+            String s = Integer.toHexString(strInt[i] ^ keyInt[i % keyInt.length]);
+            String s2 = SMSCipher.addPadding(s, 2);
+            encryptedStr.append(s2);
+        }
+        return encryptedStr.toString();
+    }
+
+    /**
+     * Decrypts the string using xor encryption with the key.
+     *
+     * @param stringToDecipher The string to be decrypted.
+     * @param key              The key used for decryption.
+     * @return The Decrypted string.
+     */
+    public static String decrypt(String stringToDecipher, String key) {
+        StringBuilder decrypted = new StringBuilder();
+        int[] keyInt = SMSCipher.fromStringToAscii(key);
+        int[] intArr = new int[stringToDecipher.length()];
+        String[] strArr = new String[stringToDecipher.length() / 2];
+        for (int i = 0; i < strArr.length; i++) {
+            strArr[i] = stringToDecipher.substring(i * 2, (i * 2) + 2);
+        }
+        for (int i = 0; i < strArr.length; i++) {
+            int decimal = Integer.parseInt(strArr[i], 16);
+            intArr[i] = decimal;
+            String s = String.valueOf(intArr[i] ^ keyInt[i % keyInt.length]);
+            int n = Integer.valueOf(s, 10);
+            decrypted.append((char) n);
+        }
+        return decrypted.toString();
+    }
+
+    /**
+     * Converts an integer to a string and adds to it the padding character.
+     * If stringToPad.length is grater or equals to @length than it only converts the intToPadd into a string.
+     *
+     * @param stringToPad The String that need the padding.
+     * @param length      The wanted length for the string.
+     * @return String with the correct length.
+     */
+    private static String addPadding(String stringToPad, int length) {
+        final char PARSING_CHARACTER = '0';
+        while (stringToPad.length() < length) {
+            stringToPad = PARSING_CHARACTER + stringToPad;
+        }
+        return stringToPad;
+    }
+
+    /**
+     * Converts every characters of the string into integer of Ascii table and save their values in an array of integer.
+     *
+     * @param stringToBeConverted The string to be converted
+     * @return An array of integer containing in position i the value of the character i of the string
+     */
+    static int[] fromStringToAscii(String stringToBeConverted) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4958670)

Reply:
Why package private? Seems like this should be used by the users, am I wrong?

Reply:
To be honest i think this is not a very useful method , you think i can also put this method private ?

Reply:
How will you test it then? If you feel like this is not its place maybe you could place it in a utils class.

Reply:
No,you're right it can't be private, i'll put it public